arch/risc-v/src/mpfs/mpfs_i2c.c

```diff
@@ -930,37 +867,44 @@ struct i2c_master_s *mpfs_i2cbus_initialize(int port)
   struct mpfs_i2c_priv_s *priv;
   int ret;
 
+#ifndef CONFIG_MPFS_COREI2C
+
   switch (port)
     {
-#ifdef CONFIG_MPFS_I2C0
+#  ifdef CONFIG_MPFS_I2C0
       case 0:
         priv = &g_mpfs_i2c0_lo_priv;
         break;
-#endif /* CONFIG_MPFS_I2C0 */
-#ifdef CONFIG_MPFS_I2C1
+#  endif /* CONFIG_MPFS_I2C0 */
+#  ifdef CONFIG_MPFS_I2C1
       case 1:
         priv = &g_mpfs_i2c1_lo_priv;
         break;
-#endif /* CONFIG_MPFS_I2C1 */
-#ifdef CONFIG_MPFS_COREI2C0
-      case 0:
-        priv = &g_mpfs_corei2c0_priv;
-        break;
-#endif
-#ifdef CONFIG_MPFS_COREI2C1
-      case 1:
-        priv = &g_mpfs_corei2c1_priv;
-        break;
-#endif
-#ifdef CONFIG_MPFS_COREI2C2
-      case 2:
-        priv = &g_mpfs_corei2c2_priv;
-        break;
-#endif
+#  endif /* CONFIG_MPFS_I2C1 */
       default:
         return NULL;
   }
 
+#else /* ifndef CONFIG_MPFS_COREI2C */
+
+  if (port < 0 || port >= CONFIG_MPFS_COREI2C_INSTANCES)
```

Comment:
The "priv->refs" logic below fails, because you unconditionally reset the locks.

Reply:
Thanks, made a straightforward attempt fix it.. just making static initialization for the mutex.

Reply:
Yep, should work now.